### PR TITLE
Batchnorm check graph

### PR DESCRIPTION
**Pull request by lixiang007666**
此PR完成了：

- [x] 打开BatchNorm的Graph测试。
- [x] 因为BatchNorm在不同状态下对running_mean和running_var有不同的动作，来源issue：https://github.com/Oneflow-Inc/oneflow/issues/7343 ，这里将oneflow到graph_train_oneflow的深拷贝提前到执行eager模式前更安全。
- [x] 优化Batchnorm的python层逻辑。

### CODE REVIEW

File: python/oneflow/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -113,13 +113,13 @@ def forward(self, x):
         if self.training:
             is_training = True
         else:
-            is_training = (self.running_mean is None) and (self.running_var is None)
+            # NOTE(lixiang): In eval mode, missing any one will select the training kernel, because in the functor layer, this check is already done.
+            is_training = (self.running_mean is None) or (self.running_var is None)
```
Comment (MARD1NO) [inline python/oneflow/nn/modules/batchnorm.py:117]: 我觉得这里还是得用and不应该用or吧，尽管functional层做了检查，functional层是保证 mean和var要么同时存在，要么同时为None。虽然你注释说的没错，但我觉得你python层检查逻辑应该和functional层一致吧

File: python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py [hunk 2]
```diff
@@ -337,21 +337,23 @@ def dual_method(self, *args, **kwargs):
                                     else:
                                         graph_kwargs[key] = copy.deepcopy(value)
 
+                            if isinstance(oneflow, flow.nn.Module):
```
Comment (strint) [inline python/oneflow/test_utils/automated_test_util/torch_flow_dual_object.py:340]: and testing_graph

File: python/oneflow/nn/modules/batchnorm.py (hunk 1 above)
Comment (strint) [inline python/oneflow/nn/modules/batchnorm.py:117]: 这里evaluate的时候，我们是不允许其为空，还是现在这种容错的方式呢？
也就是 track_running_stats 为false时，我们是否还允许其做evaluate呢？

File: python/oneflow/nn/modules/batchnorm.py [hunk 3]
```diff
@@ -113,13 +113,13 @@ def forward(self, x):
...
+        # NOTE(lixiang): If it is training mode, pass running_mean and running_var directly to the functor layer.
         return flow._C.normalization(
             x,
-            self.running_mean
-            if not self.training or self.track_running_stats
-            else None,
-            self.running_var if not self.training or self.track_running_stats else None,
+            self.running_mean if self.track_running_stats else None,
+            self.running_var if self.track_running_stats else None,
```
Comment (strint) [inline python/oneflow/nn/modules/batchnorm.py:122]: 这里不用判断，因为track_running_stats为false时，本来就是None

**Review by strint — APPROVED** [review]

**Review by MARD1NO — APPROVED** [review]
